### PR TITLE
CLD-7841 Optimize E2E testing costs

### PR DESCRIPTION
**Pull request by mvitale1989**
#### Summary

This PR brings various improvements to our E2E testing setup:
- Move the E2E test runs to the standard-tier, Github-hosted runners
- Record and store videos of failing testcase runs
- Upload single run summary artifact at the end of the run (including above-mentioned videos)
  * Required isolating cypress and server logfiles per-worker
- Notify user on the PR when the run has has finished
- Reduce RollingRelease tests' parallelization to 10, to reduce the impact of the initial smoketests
- Merge `start-server` step with `prepare-server`
- Upgrade Cypress
- Upgrade Keycloak version

The slightly slower instance caused a few E2E test cases to fail, most of which have been fixed:
```
integration/channels/accessibility/accessibility_post_spec.js
  ^ There's the extra `actions` button now, that we need to account for.
integration/channels/archived_channel/post_menu_spec.ts
  ^ Fixed how we initiate the header change.
integration/channels/custom_status/custom_status_expiry/custom_status_expiry_1_spec.ts
  ^ Fixed flakiness: under very unfortunate timing, there may be a difference between the status expiry time between the testcase code and the server computation.
integration/channels/files_and_attachments/youtube_video_spec.js
  ^ More robust syncrhonization to frontend updates
integration/channels/integrations/message_to_channel_via_slash_command_spec.js
  ^ More robust syncrhonization to frontend updates
integration/channels/system_console/user_management_spec.js
  ^ More robust syncrhonization to frontend updates
integration/channels/bot_accounts/display_name_spec.js
  ^ More robust syncrhonization to frontend updates
integration/channels/system_console/user_management/users_spec.js
  ^ More robust syncrhonization to frontend updates
integration/channels/system_console/user_management_not_cloud_spec.js
  ^ More robust syncrhonization to frontend updates
integration/playbooks/channels/slash_command/test_spec.js
  ^ More robust synchronization to server config updates
integration/channels/system_console/unsaved_changes_spec.js
  ^ Increase state checking timeouts
integration/channels/integrations/builtin_commands/common_commands_1_spec.js
  ^ The `cy.wait()` is very time sensitive, especially because the utility functions already introduce some delay.
integration/channels/integrations/builtin_commands/invite_command_spec.js
  ^ Fix GM sending, fix Cypress writing to the right channel after GM switch
integration/playbooks/channels/rhs/status_update_spec.js
  ^ Avoid the slash command autocomplete popup from hogging on the 'Enter' key, preventing the slash commands from being posted
integration/playbooks/channels/slash_command/commands_spec.js
  ^ Avoid the slash command autocomplete popup from hogging on the 'Enter' key, preventing the slash commands from being posted
integration/playbooks/playbooks/list_spec.js
  ^ The playbooks window is considered partially hidden by cypress, so we have to force the drag-drop
```

Some testcases seem to fail due to an actual quality issue:
```
integration/channels/enterprise/accessibility/accessibility_modals_dialogs_spec.js
  ^ Can reproduce even outside of Cypress: open channel -> click on `Add people` and type `u` as fast as possible -> no users (note that the API call _does_ happen, possibly a frontend issue)
integration/channels/toast/unread_with_bottom_start_toast_spec.js
  ^ Can reproduce even outside of Cypress: create channel -> post a lot -> set as unread, toast is not there -> change channel -> go back, toast still not there
integration/channels/mark_as_unread/toast_appears_unread_spec.js
  ^ Can reproduce even outside of Cypress
integration/channels/scroll/deleting_scroll_spec.js
  ^ Can reproduce in Cypress
```

A few more testcases seem either random, or I couldn't reproduce them:
```
integration/channels/messaging/message_reply_spec.js
integration/channels/notifications/at_mentions_spec.js
  ^ Notification aren't working at all on my local cypress runs
integration/channels/system_console/site_url_config_spec.js
  ^ This seems random: the frontend tries to access `/admin_console/environment/web_server`, but suddenly the page switches to `/admin-console/reporting/workspace_optimization`.
  ^ See the artifact from this run: https://github.com/mattermost/mattermost/actions/runs/10376547866
```

#### Ticket Link

https://mattermost.atlassian.net/browse/CLD-7841
https://mattermost.atlassian.net/browse/CLD-5784

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/e2e-fulltests-ci.yml [hunk 1]
```diff
@@ -229,3 +240,34 @@ jobs:
...
+    steps:
+      - name: ci/notify-user-test-completion
+        run: |
+          if [ -n "$PR_NUMBER" ]; then
+            gh issue -R "${{ github.repository }}" comment "$PR_NUMBER" --body-file - <<EOF
+          E2E test has completed for commit \`${COMMIT_SHA}\`${MM_ENV:+, with \`MM_ENV=$MM_ENV\`}.
+          You can check its result on the [Automation Dashboard]($CYCLE_URL), or by checking out the status check named \`${STATUS_CHECK_CONTEXT}\` in this PR.
+          The run summary artifact is available in the corresponding [Workflow Run]($WORKFLOW_RUN_URL).
+          EOF
+          fi
```
Comment (yasserfaraazkhan) [inline .github/workflows/e2e-fulltests-ci.yml:273]: 💯

File: e2e-tests/README.md [hunk 2]
```diff
@@ -55,7 +55,7 @@ Notes:
...
-- `make start-server` won't cleanup containers that don't change across runs. This means that you can use it to emulate a Mattermost server upgrade while retaining your database data by simply changing the `SERVER_IMAGE` variable on your machine, and then re-runing `make start-server` (and `make prepare-server` if you intend to run cypress again, since its container is also recreated). But this also means that if you want to run a clean local environment, you may have to manually run `make stop` to cleanup any running containers and their volumes, which include e.g. the database.
+- `make start-server` won't cleanup containers that don't change across runs. This means that you can use it to emulate a Mattermost server upgrade while retaining your database data by simply changing the `SERVER_IMAGE` variable on your machine, and then re-runing `make start-server`. But this also means that if you want to run a clean local environment, you may have to manually run `make stop` to cleanup any running containers and their volumes, which include e.g. the database.
```
Comment (saturninoabril) [inline e2e-tests/README.md:58]:
```suggestion
- `make start-server` won't cleanup containers that don't change across runs. This means that you can use it to emulate a Mattermost server upgrade while retaining your database data by simply changing the `SERVER_IMAGE` variable on your machine, and then re-running `make start-server`. But this also means that if you want to run a clean local environment, you may have to manually run `make stop` to cleanup any running containers and their volumes, which include e.g. the database.
```

**Review by saturninoabril — APPROVED** [review]
Wow, lots of good things here! Huge thanks!

**Review by yasserfaraazkhan — APPROVED** [review]

File: server/build/gitlab-dc.common.yml [hunk 3]
```diff
@@ -61,18 +61,22 @@ services:
       timeout: 15s
       retries: 12
   keycloak:
-    image: mattermostdevelopment/mirrored-keycloak:10.0.2
+    image: quay.io/keycloak/keycloak:23.0.7
```
Comment (saturninoabril) [inline server/build/gitlab-dc.common.yml:64]: @mvitale1989 I noticed this late. Is there a plan to host this image in our `mattermostdevelopment` dockerhub?

Comment (mvitale1989) [reply server/build/gitlab-dc.common.yml:64]: The medium term plan would be to completely remove the `gitlab-dc.common.yml` file, in favour of using the `docker-compose.yml` file, see [this ticket](https://mattermost.atlassian.net/browse/CLD-8147), so it's probably not worth mirroring this keycloak image for now.
But it's worth understanding if we won't get into any Docker Hub limits as part of this migration, i'll add a note to the ticket

**Pull request merged**